cirq/google/programs.py

```diff
@@ -351,22 +351,22 @@ def raise_missing_fields(gate_name: str):
 
 
 def _parameterized_value_from_proto_dict(message: Dict
-                                         ) -> Union[value.Symbol, float]:
+                                         ) -> Union[sympy.Symbol, float]:
     parameter_key = message.get('parameter_key', None)
     if parameter_key:
-        return value.Symbol(parameter_key)
+        return sympy.Symbol(parameter_key)
     if 'raw' in message:
         return message['raw']
     raise ValueError('No value specified for parameterized float. '
                      'Expected "raw" or "parameter_key" to be set. '
                      'message: {!r}'.format(message))
 
 
-def _parameterized_value_to_proto_dict(param: Union[value.Symbol, float]
+def _parameterized_value_to_proto_dict(param: Union[sympy.Symbol, float]
                                        ) -> Dict:
     out = {}  # type: Dict
-    if isinstance(param, value.Symbol):
-        out['parameter_key'] = param.name
+    if isinstance(param, sympy.Basic):
```

Comment:
This is the wrong type check at this point. We only want `sympy.Symbol`. If it's a `sympy.Basic` but not a `sympy.Symbol`, raise a `NotImplementedError` explaining we don't support serializing it yet.